### PR TITLE
fix(tools)!: removes the esbuild plugins and helpers

### PR DESCRIPTION
**Pull request by bennypowers**
## What I did
- Removed esbuild helpers from tools
- moved test runner, dev server and cem/a helpers to their respective folders
- update tools readme
- mark `pfeCustomElementsManifestConfig` as deprecated so we can remove it in a subsequent release without breaking
### Why?
The esbuild helpers had gotten more complex and had built in assumptions which no longer obtained. rather than prescribing build scripts to try and cover every case for every user, it's much better for users to write their own esbuild scripts or rollup configs. furthermore, the esbuild API is in flux, so maintaining these plugins as part of our public API opens us up to maintenance burden and our users to dependency churn.

Our cem/a config relies on a series of patches to an out of date version of cem/a. We should not recommend these in our public api. Better to remove them, and the dependency on cem/a, and either fork cem/a or use the upcoming lit-labs analyzer

It's better to remove these now rather than in a breaking change after RC

### CODE REVIEW

File: tools/pfe-tools/README.md [hunk 1]
```diff
@@ -7,57 +7,21 @@ Tools and utilities for building PatternFly Elements and other design systems.
...
-
-Helpers for working with esbuild in design system monorepos
+- Helpers for collating and rendering [custom-elements manifests][cem] in 11ty
+  sites.
+- Various 11ty utility plugins
 
 ## test
 
-Helpers for testing web components using [web test runner](https://modern-web.dev/docs/test-runner/overview/)
+Helpers for testing web components using [web test runner][wtr]
```
Comment (zeroedin) [inline tools/pfe-tools/README.md:20]: TIL:  silly, but I didn't know you could reference links like this in markdown.

**Comment by brianferry** [comment]
Builds & demos all looking good from here, readme for tools looks good here too

File: tools/pfe-tools/test/config.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
-import type { PfeDevServerConfigOptions } from './dev-server';
 import type { TestRunnerConfig } from '@web/test-runner';
 
 import { playwrightLauncher } from '@web/test-runner-playwright';
 import { summaryReporter, defaultReporter } from '@web/test-runner';
 import { junitReporter } from '@web/test-runner-junit-reporter';
 import { a11ySnapshotPlugin } from '@web/test-runner-commands/plugins';
 
-import { pfeDevServerConfig } from './dev-server.js';
+import { pfeDevServerConfig, type PfeDevServerConfigOptions } from '../dev-server/config.js';
```
Comment (zeroedin) [inline tools/pfe-tools/test/config.ts:8]: Another TIL:  I was unaware you could add a type to a regular import line, very useful for keeping things concise.

**Review by zeroedin — CHANGES_REQUESTED** [review]
`npm run new` fails.  Needs a `wireit` command in package.json.  Will submit commit.

```json
"new": "wireit",
...
"new": {
  "command": "./tools/create-element/bin/main.js",
  "dependencies": [
    "build:create"
  ]
},
```

File: package.json [hunk 3]
```diff
@@ -225,6 +225,12 @@
         "build:icons"
       ]
     },
+    "new": {
+      "command": "./tools/create-element/bin/main.js",
+      "dependencies": [
+        "build:create"
```
Comment (brianferry) [inline package.json:231]: Looks like `npm run new` is hanging after the `What is the element's name?` message

**Review by zeroedin — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (zeroedin) [reply package.json:232]: This fix definitely works, we couldn't figure out why the chaining `&&` was working but the `dependencies` in wireit command weren't working in a similar fashion.  Nice fix brian.

**Comment by bennypowers** [comment]
Thanks @brianferry and @zeroedin. Like the fix!

**Pull request merged**